### PR TITLE
deps: downgrade Netty due to incompatibility with grpc-java

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

Netty 4.1.111.Final is incompatible with grpc-java < 1.65.0 and >= 1.51.0. It can introduce buffer corruption at any point, which if you're lucky is detected, but may also go unnoticed and result in garbage and waste of time.

Wait until grpc-java releases a compatible version before updating to 4.1.111.Final. Note that this also blocks Spring Boot 3.3.1 which depends on Netty 4.1.111.Final.

See:

- https://github.com/grpc/grpc-java/issues/11284#issuecomment-2168264887
- https://github.com/grpc/grpc-java/releases/tag/v1.64.0

## Related issues

blocks #19581
blocks #19265
blocks #19264 
blocks #19259
blocks #19258

### CODE REVIEW

**Comment by npepinpe** [comment]
I'm now wondering if this may not have caused more issues than we thought, considering the Netty issue implies corruption can go unnoticed until it hits your application, and this was merged on the 11th of June :thinking:

**Comment by Zelldon** [comment]
@npepinpe do we need to configure renovate here ? 🤔

**Comment by npepinpe** [comment]
Maybe? Do you know how to pause for a specific dep? I'm no renovate expert :s

**Comment by Zelldon** [comment]
I think it is similar to something like https://github.com/camunda/camunda/blob/main/.github/renovate.json#L50 

But do you then keep an eye on the updates so we don't miss it ? 🤔

**Comment by Zelldon** [comment]
Regarding renovate we can also ask our expert (summon @cmur2 👻 )

**Comment by Zelldon** [comment]
What you where writing regarding the corruption and 

> DO NOT USE NETTY 4.1.111 WITH GRPC. It appears there is corruption. There's no guarantees the corruption would be detected.

Looks really scary 😅

**Review by ChrisKujawa — APPROVED** [review]

File: .github/renovate.json
Comment (cmur2) [inline .github/renovate.json:74]: Please add a description (text + e.g. ticket link) like many of the rules have :)

File: .github/renovate.json
Comment (cmur2) [inline .github/renovate.json:76]: Should work https://docs.renovatebot.com/configuration-options/#allowedversions

You'd also have to backport this to `stable/**` branches for full effect since PRs are open there as well: https://github.com/camunda/camunda/pull/19258

Alternatively you can always manually close such a PR to ignore that specific version, instead of adding a Renovate rule

Comment (npepinpe) [reply .github/renovate.json:76]: The PRs are blocked on stable branches explicitly, so I think it's fine not to backport for now.

File: .github/renovate.json
Comment (npepinpe) [reply .github/renovate.json:75]: I thought about adding a comment but remembered JSON has no comments, I didn't know about the "description" field (but should've since it's used elsewhere :see_no_evil:)

**Comment by juliojgd** [comment]
@npepinpe You've banned Netty 4.1.111 version but, if you read the full conversation at grpc-java, you will realize that there is nothing wrong with Netty 4.1.111 but the problem is with grpc-java. You should upgrade to Netty 4.1.111 when grpc-java releases the fixed versions (and you adopted these new grpc-java fixed versions).

**Comment by Zelldon** [comment]
> @npepinpe You've banned Netty 4.1.111 version but, if you read the full conversation at grpc-java, you will realize that there is nothing wrong with Netty 4.1.111 but the problem is with grpc-java. You should upgrade to Netty 4.1.111 when grpc-java releases the fixed versions (and you adopted these new grpc-java fixed versions).

I think that is the plan. :) but thanks for raising 👍🏼

**Comment by npepinpe** [comment]
As @Zelldon mentioned, that's exactly our plan :) We're just waiting for the 1.65 release of grpc-java in order to update.

**Pull request merged**
